File: dashboard/src/types/scan.ts
export type Decision = 'LIBERADO' | 'VERIFICAR' | 'INCONCLUSIVO'

export interface Detection {
  class_name: string
  confidence: number
  bbox: [number, number, number, number]
}

export interface ScanResult {
  id: string
  created_at: string
  filename: string
  decision: Decision
  detections: Detection[]
  processing_time_ms: number | null
}

File: api/schemas/scan.py
from datetime import datetime
from pydantic import BaseModel

from core.rules import Decision


class DetectionSchema(BaseModel):
    class_name: str
    confidence: float
    bbox: list[float]


class ScanResult(BaseModel):
    id: str
    created_at: datetime
    filename: str
    decision: Decision
    detections: list[DetectionSchema]
    processing_time_ms: float | None

    class Config:
        from_attributes = True


class FeedbackRequest(BaseModel):
    operator_id: str
    feedback: str  # "confirmed" | "false_positive" | "false_negative"


class ScanListItem(BaseModel):
    id: str
    created_at: datetime
    filename: str
    decision: Decision
    operator_feedback: str | None

    class Config:
        from_attributes = True
